examples/simple-generator/src/app/index.ts

```diff
@@ -0,0 +1,142 @@
+import { join } from 'path';
+import Generator from 'yeoman-generator';
+
+import type { Template, LROPSettings, OVPSettings } from '@sap-ux/fiori-elements-writer';
+import { generate as generateFE, OdataVersion, TemplateType } from '@sap-ux/fiori-elements-writer';
+import type { OdataService } from '@sap-ux/odata-service-writer';
+import type { Template as FreestyleTemplate } from '@sap-ux/fiori-freestyle-writer';
+import { generate as generateUI5, TemplateType as FreestyleTemplateType } from '@sap-ux/fiori-freestyle-writer';
+import type { Ui5App } from '@sap-ux/ui5-application-writer';
+import { isAppStudio } from '@sap-ux/btp-utils';
+import type { ServiceInfo } from './service';
+import { getServiceInfo, getServiceInfoInBAS } from './service';
+
+export default class extends Generator {
+    private app!: Ui5App & { app: { flpAppId: string } };
+    private service!: OdataService;
+    private template: {
+        type: TemplateType | FreestyleTemplateType.Basic;
+        settings?: LROPSettings | OVPSettings | {};
+    };
+
+    initializing(): void {
+        this.log(
+            'Example of a simple Fiori elements for OData v4 generator that only creates listreport objectpage applications.'
+        );
+    }
+
+    async prompting(): Promise<void> {
+        const { name, template } = await this.prompt([
+            {
+                type: 'input',
+                name: 'name',
+                message: 'Application name',
+                validate: (answer) => !!answer
+            },
+            {
+                type: 'list',
+                name: 'template',
+                message: 'Template',
+                choices: [
+                    { value: TemplateType.ListReportObjectPage, name: 'List Report' },
+                    { value: TemplateType.OverviewPage, name: 'Overview Page' },
+                    { value: FreestyleTemplateType.Basic, name: 'Freestyle App' }
+                ]
+            }
+        ]);
+
+        let service: ServiceInfo;
+        if (isAppStudio()) {
+            service = await getServiceInfoInBAS(this);
+        } else {
+            service = await getServiceInfo(this);
+        }
+
+        const { entity } = await this.prompt({
+            type: 'input',
+            name: 'entity',
+            message: 'Main entity',
+            validate: (answer) => !!answer
+        });
+
+        this.service = {
+            version: OdataVersion.v2,
```

Comment:
Does the version matter?

Reply:
Yeah, the example would be a bit more complicated if we support any OData version because then I would also have to check the metadata of the service and derive the version from it. It is, of course, doable but I wanted to keep the generator as simple as possible.